File: DataStructures/flattenNestedDict.py
# def flattenNestedDict(d):
#     return
# print(flattenNestedDict({
#     "key": 3,
#     "foo": {
#         "a": 5,
#         "bar": {
#             "baz": 8
#         }
#     }
# }))

d = {
    'name': 'james',
    'age': 20,
    'course':  'maths computer',
    'kk': {
        'k1': 'blunt',
        'k2': {
            'swishers': 'kid'
        }
    }
}

for k, v in d.items():
    print([k, v])

File: js-datastructures/FlatenedNestedDict/flattenNestedDict.js
const flattenNestedDict = dict => {
  const res = {};
  for (const key in dict) {
    if (typeof dict[key] === "object" && dict[key] !== null) {
      let flattened = flattenNestedDict(dict[key]);
      for (const nestedKey in flattened) {
        res[key + "." + nestedKey] = flattened[nestedKey];
      }
    } else {
      res[key] = dict[key];
    }
  }
  return res;
};
let d = {
  key: 3,
  foo: {
    a: 5,
    bar: {
      baz: 8
    }
  }
};
module.exports = flattenNestedDict;
console.log(flattenNestedDict(d));
